colorcoder/Main.java

```diff
@@ -1,120 +1,28 @@
-                }
-            }
-            return null;
-        }
-    };
 
-    final static String MajorColorNames[] = {
-        "White", "Red", "Black", "Yellow", "Violet"
-    };
-    final static int numberOfMajorColors = MajorColorNames.length;
-    final static String MinorColorNames[] = {
-        "Blue", "Orange", "Green", "Brown", "Slate"
-    };
-    final static int numberOfMinorColors = MinorColorNames.length;
+	public static void main(String[] args) {
+		testNumberToPair(4, MajorColor.WHITE, MinorColor.BROWN);
+		testNumberToPair(5, MajorColor.WHITE, MinorColor.SLATE);
 
-    static class ColorPair {
-        private MajorColor majorColor;
-        private MinorColor minorColor;
-        
-        public ColorPair(MajorColor major, MinorColor minor)
-        {
-            majorColor = major;
-            minorColor = minor;
-        }
-        public MajorColor getMajor() {
-            return majorColor;
-        }
-        public MinorColor getMinor() {
-            return minorColor;
-        }
-        String ToString() {
-            String colorPairStr = MajorColorNames[majorColor.getIndex()];
-            colorPairStr += " ";
-            colorPairStr += MinorColorNames[minorColor.getIndex()];
-            return colorPairStr;
-        }
-    };
+		testPairToNumber(MajorColor.BLACK, MinorColor.ORANGE, 12);
+		testPairToNumber(MajorColor.VIOLET, MinorColor.SLATE, 25);
+	}
 
-    static ColorPair GetColorFromPairNumber(int pairNumber) {
-        int zeroBasedPairNumber = pairNumber - 1;
-        MajorColor majorColor = 
-            MajorColor.fromIndex(zeroBasedPairNumber / numberOfMinorColors);
-        MinorColor minorColor =
-            MinorColor.fromIndex(zeroBasedPairNumber % numberOfMinorColors);
-        return new ColorPair(majorColor, minorColor);
-    }
-    static int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
-        return major.getIndex() * numberOfMinorColors + minor.getIndex() + 1;
-    }
+	static void testNumberToPair(int pairNumber, MajorColor expectedMajor, MinorColor expectedMinor) {
+		ColorPair colorPair = new ColorPair(expectedMajor, expectedMinor);
+		colorPair = colorPair.getColorFromPairNumber(pairNumber);
+		System.out.println("Got pair " + colorPair.toString());
+		assert (colorPair.getMajor() == expectedMajor);
+		assert (colorPair.getMinor() == expectedMinor);
+	}
 
-    static void testNumberToPair(int pairNumber,
-        MajorColor expectedMajor,
-        MinorColor expectedMinor)
-    {
-        ColorPair colorPair = GetColorFromPairNumber(pairNumber);
-        System.out.println("Got pair " + colorPair.ToString());
-        assert(colorPair.getMajor() == expectedMajor);
-        assert(colorPair.getMinor() == expectedMinor);
-    }
+	static void testPairToNumber(MajorColor major, MinorColor minor, int expectedPairNumber) {
+		ColorPair colorPair = new ColorPair(major, minor);
+		int pairNumber = colorPair.getPairNumberFromColor(major, minor);
```

Comment:
The major and minor used in line 22 to create the colorPair has no relation to the ones passed to the method in line 23. If color pair had the major and minor set as class members, then the API should be as follows,
**colorPair.getPairNumberFromColor()**